### PR TITLE
add SVG capabilities

### PR DESCRIPTION
**Pull request by thomasleveil**
This is a quick implementation of SVG support.

In it's current state it does produce a SVG file which can at least be displayed in Inkscape. This is my first time playing with SVG, so there must be things to improve in the produced document format.

I hope this is enough to get more SVG savvy people involved.

----

Example of produced svg file 
[signatures.zip](https://github.com/gcacace/android-signaturepad/files/143668/signatures.zip)

### CODE REVIEW

**Comment by thomasleveil** [comment]
Ideas of svg optimizations: https://jakearchibald.github.io/svgomg/

File: SignaturePad-Example/src/main/java/com/github/gcacace/signaturepad/MainActivity.java [hunk 1]
```diff
@@ -106,4 +111,21 @@ public boolean addSignatureToGallery(Bitmap signature) {
         }
         return result;
     }
+
+    public boolean addSignatureToGallery(String signatureSvg) {
```
Comment (gcacace) [inline SignaturePad-Example/src/main/java/com/github/gcacace/signaturepad/MainActivity.java]: Please rename this function in something like `addSvgSignatureToGallery`, and rename also the other function generating the JPG/PNG image in `addJpgSignatureToGallery`.

File: signature-pad/src/main/java/com/github/gcacace/signaturepad/utils/Svg.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.github.gcacace.signaturepad.utils;
+
+public class Svg {
```
Comment (gcacace) [inline signature-pad/src/main/java/com/github/gcacace/signaturepad/utils/Svg.java]: What do you think about renaming this class in something like `SvgCurve`, then create an instance of it and store inside it the list of `<path/>`, finally create a `build()` method to apply header, footer and return the SVG XML.

File: signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java [hunk 3]
```diff
@@ -33,6 +34,8 @@
     private float mLastWidth;
     private RectF mDirtyRect;
 
+    private final StringBuilder mSvgBuilder = new StringBuilder();
```
Comment (gcacace) [inline signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java]: I would prefer having an instance of `SvgCurve` instead.

File: signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java [hunk 4]
```diff
@@ -199,6 +203,12 @@ public boolean isEmpty() {
         return mIsEmpty;
     }
 
+    public String getSignatureSvg() {
+        int width = getTransparentSignatureBitmap().getWidth();
+        int height = getTransparentSignatureBitmap().getHeight();
+        return Svg.header(width, height) + mSvgBuilder.toString() + Svg.footer();
```
Comment (gcacace) [inline signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java]: In that case, as per my previous comments, here you should call `svgCurve.build()` to get the SVG XML string.

**Comment by thomasleveil** [comment]
Thank you the code review and valuable feedbacks :memo:

last commit introduces a new `SvgBuilder` class in which all SVG related code is enclosed. This is the place where further optimizations on the produced SVG document can take place without fearing regressions on the rest of the project.

Please let me know if you want me to squash those commits

**Comment by gcacace** [comment]
Thanks for your commits! If you can, please squash all in just a single commit so I can merge with master.

**Comment by thomasleveil** [comment]
squashed. I left out the first commit which has nothing to do with SVG and added two commit to illustrate how one can contribute by optimizing the produced SVG document.

**Pull request merged**